### PR TITLE
Expand checkPot to warn about messages which were marked as not having translator comments but are no longer present in the translations template

### PR DESCRIPTION
**Pull request by lukaszgo1**
### Link to issue number:
None
### Summary of the issue:
Currently `checkPot` has a list of messages which are translatable but don't have commends for translators. While it warns about messages in this list for which comments are added it does not warn about these which are no longer present in the .pot template.

### Description of how this pull request fixes the issue:
I've modified `checkPot` so that if there are messages in `EXPECTED_MESSAGES_WITHOUT_COMMENTS` which were removed from the source code it lists them and fails similarly to the case in which a new message has no comments for translators.
### Testing strategy:
Ensured that 12 messages which are in `EXPECTED_MESSAGES_WITHOUT_COMMENTS` which are not in source code any longer were properly flagged and the build failed both locally and on AppVeyor [build log here](https://ci.appveyor.com/project/NVAccess/nvda/builds/40203327).
### Known issues with pull request:
None known
### Change log entries:
None needed - this just improves our tests.

### Code Review Checklist:

- [X] Pull Request description is up to date.
- [X] Unit tests.
- [X] System (end to end) tests.
- [X] Manual testing.
- [X] User Documentation.
- [X] Change log entry.
- [X] Context sensitive help for GUI changes.
- [X] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers

### CODE REVIEW

File: tests/checkPot.py [hunk 1]
```diff
@@ -203,9 +197,19 @@ def checkPot(fileName):
 					"Message: {message}\n"
 					.format(error=error, lines=" ".join(sourceLines), message=message))
 				continue
-	print("{errors} errors, {unexpectedSuccesses} unexpected successes, {expectedErrors} expected errors"
-		.format(errors=errors, unexpectedSuccesses=unexpectedSuccesses, expectedErrors=expectedErrors))
-	return errors + unexpectedSuccesses
+	removedTranslatableMessages = EXPECTED_MESSAGES_WITHOUT_COMMENTS - foundMessagesWithOutComments
+	if removedTranslatableMessages:
+		print(
+			"The following messages arre no longer present in the source code "
```
Comment (seanbudd) [inline tests/checkPot.py:203]:
```suggestion
			"The following messages are no longer present in the source code "
```

File: tests/checkPot.py [hunk 2]
```diff
@@ -203,9 +197,19 @@ def checkPot(fileName):
...
+	removedTranslatableMessages = EXPECTED_MESSAGES_WITHOUT_COMMENTS - foundMessagesWithOutComments
+	if removedTranslatableMessages:
+		print(
+			"The following messages arre no longer present in the source code "
+			"and should be removed from `EXPECTED_MESSAGES_WITHOUT_COMMENTS`:"
+		)
+		print('\n'.join(removedTranslatableMessages))
+	print(
+		f"{errors} errors, {unexpectedSuccesses} unexpected successes, {expectedErrors} expected errors, "
+		f"{len(removedTranslatableMessages)} messages marked as expected failures removed from the source code"
```
Comment (seanbudd) [inline tests/checkPot.py:209]:
```suggestion
		f"{len(removedTranslatableMessages)} messages marked as expected errors not present in the source code"
```

**Review by seanbudd — CHANGES_REQUESTED** [review]
Thanks @lukaszgo1, this is a helpful check. I've made a suggestion to rephrase a message, and fix a typo. Would you also be able to add a unit test for this check, like the other checkPot tests? It would also be good if the other tests have `len(removedTranslatableMessages) == 0`, so that only one test fails if the check is inaccurate.

**Comment by lukaszgo1** [comment]
> 
> 
> Thanks @lukaszgo1, this is a helpful check. I've made a suggestion to rephrase a message, and fix a typo. 

Thanks for these - applied now.


> Would you also be able to add a unit test for this check, like the other checkPot tests? 

This is being tested implicitly by all other tests. I'm not really sure if adding separate tests for aspects of `checkPot` which are going to be removed as soon as all messages in the source have comments for translators is a good use of anyone's time.

> It would also be good if the other tests have `len(removedTranslatableMessages) == 0`, so that only one test fails if the check is inaccurate.


It is of course doable but it would require a complete rewrite of the test data and therefore each and every unittest for the `checkPot`  which I'm not sure is a good direction to take - see above w.r.t parts of `checkPot` being hopefully temporary.

**Review by seanbudd — APPROVED** [review]
That is fair, in the meantime these aspects will be tested implicitly. 
Thanks again for your work here.

**Pull request merged**